### PR TITLE
Command list (to resolve #33)

### PR DESCRIPTION
**Pull request by brittanydionigi**
Soul-crushing diff, sorry. Breaks out our commands to separate files, but takes away our aliased commands `create`, `g`, `scaffold`, `serve`. Need to figure out how best to implement those. Opening this now for moar eyes.

### CODE REVIEW

File: commands/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+var fs = require('fs');
+var path = require('path');
+
+var commandFiles = fs.readdirSync(__dirname);
+
+
+commandFiles.forEach(function(file) {
+  var commandName = path.basename(file, '.js');
+  if (commandName === "index") { return; }
```
Comment (tofumatt) [inline commands/index.js]: I guess we should add it to the JS style checker, but I think most of Moz (myself included) prefers to keep even single-line blocks on their own line, for future reference. Makes things a bit easier to scan.

**Comment by brittanydionigi** [comment]
So ember-cli does a nice job organizing their commands that makes it easy to add/update as you go along. It looks like their implementing them as constructors where they can each set their own alias property, though this also makes the check for finding the command a little trickier. We don't have nearly half the complexity of ember-cli at the moment...so I'd like to find a simpler solution for now if possible. There could also be one right in front of my face that I'm not seeing because I have been staring for too long and now I am cross-eyed.

**Comment by tofumatt** [comment]
Hah, yeah, I'm also a bit tired at this point and can't quite make sense of how to tweak some things, but I see where this is going and organizationally like it a _lot_ more than the spaghetti soup I can constructed.

I'll eye it more tomorrow, but I think this is the right plan. :+1:

**Comment by brittanydionigi** [comment]
This has been updated a bit. It now takes care of aliased commands by adding them as properties to the constructed command.  (as seen here... https://github.com/mozilla/recroom/blob/command-list/commands/new.js) 

This set up is particularly nice, because when we go to create a help menu for `recroom --help`, we can just pull all the info for each command from these properties.

Still not a huge fan of the way we are looking up the user's command: https://github.com/mozilla/recroom/blob/command-list/recroom.js#L46 (Cycling through each of the commands, and checking their aliases) but I think it winds up being the least work for us maintenance-wise as we add or update commands.

**Comment by brittanydionigi** [comment]
Also updated to take care of running the build process after creating a new application. (Issue here: #47 )

**Comment by tofumatt** [comment]
Coolio... is this still WIP or is it complete in your eyes? I'm gonna check it out and run it locally now, but let me know if I should hold off on considering it merge-able.

**Comment by brittanydionigi** [comment]
All good in my eyes. Play around with creating a new project and using the aliases like g/scaffold for generating something.

(Also I can clean up the commit history if you want. It's not out of control, but, there's some squashing that could be done.)

File: command-model.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+module.exports = Command;
+
+function Command(name) {
+    this.binaryPath = __dirname + '/node_modules/.bin/';
+    this.aliases = this.aliases || [];
```
Comment (tofumatt) [inline command-model.js:7]: I like this, very cool :+1:

**Comment by tofumatt** [comment]
This looks great to me; just tested it out (oops, late!) and it works great too. I'm just gonna merge it; we can worry about nicer commit histories when we're a bit less alpha :smile:

**Pull request merged**